packages/amplify-graphql-index-transformer/src/resolvers/resolvers.ts

```diff
@@ -51,10 +51,11 @@ const API_KEY = 'API Key Authorization';
  * replaceDdbPrimaryKey
  */
 export function replaceDdbPrimaryKey(config: PrimaryKeyDirectiveConfiguration, ctx: TransformerContextProvider): void {
+  const useAmplifyManagedTableResources: boolean = ctx.transformParameters.useAmplifyManagedTableResources;
   // Replace the table's primary key with the value from @primaryKey
   const { field, object } = config;
   const table = getTable(ctx, object) as any;
-  const cfnTable = table.table;
+  const cfnTable = useAmplifyManagedTableResources ? table.node.defaultChild.node.defaultChild : table.table;
```

Comment:
We do this `table.node.defaultChild...` dance a few places, but it's pretty unintuitive. Can we centralize that into a helper method so it's clearer what you're doing?

Reply:
The new table's path is `AmplifyDynamoDBTable -> CustomResource -> CfnCustomResource`, which needs two `defaultChild` call. The original structure is `Table -> CfnTable`, where `table` is the attribute refers to the `CfnTable`. I can use a helper function and comments to clarify this in the follow-up PRs.